docs/contributors/principles.md

```diff
@@ -0,0 +1,33 @@
+# Principles of Minikube
+
+The primary goal of minikube is to make it simple to run Kubernetes on your local machine, both for getting started and day-to-day development workflows.
+
+Here are some specific features that align with our goal:
+
+* Single command setup and teardown UX.
+* Support most portable Kubernetes core features (local storage, networking, auto-scaling, loadbalancing, etc.)
+* Unified UX across OSes.
+* Minimal dependencies on third party software.
+* Minimal resource overhead.
+* Becoming the default local-cluster setup for Kubernetes
+
+## Non-Goals
+* Simplifying Kubernetes production deployment experience. Kube-deploy is attempting to tackle this problem.
```

Comment:
we should remove kube-deploy - or refer to "kubeadm, cluster API and other projects" instead.